lib/qmp.ml

```diff
@@ -143,10 +147,22 @@ let message_of_string str =
   in
 
   let event json =
-    let ts x = json |> U.member "timestamp" |> U.member x |> U.to_int in
-    { timestamp = (ts "seconds", ts "microseconds")
-    ; event     = json |> U.member "event" |> U.to_string
-    }
+    let event' json =
+      let ts x = json |> U.member "timestamp" |> U.member x |> U.to_int in
+      { timestamp = (ts "seconds", ts "microseconds")
+      ; event     = json |> U.member "event" |> U.to_string
+      }
+    in
+    let rtc_offset json =
+      json |> U.member "data" |> U.member "offset" |> U.to_int
+    in
+    let event_data = match json |> U.member "event" |> U.to_string with
```

Comment:
This could be simplified a bit: you are doing already the match above.
You could do `json |> U.member "event" |> U.to_string` before anything else and store it in a variable, then use it to the functions below.

Reply:
Thanks, updated